spirv_msl.cpp

```diff
@@ -1434,9 +1446,32 @@ void CompilerMSL::mark_as_packable(SPIRType &type)
 }
 
 // If a vertex attribute exists at the location, it is marked as being used by this shader
-void CompilerMSL::mark_location_as_used_by_shader(uint32_t location, StorageClass storage)
+void CompilerMSL::mark_location_as_used_by_shader(uint32_t location, const SPIRType &type, StorageClass storage)
 {
-	if ((get_execution_model() == ExecutionModelVertex || is_tessellation_shader()) && (storage == StorageClassInput))
+	if ((get_execution_model() != ExecutionModelVertex && !is_tessellation_shader()) || (storage != StorageClassInput))
```

Comment:
This is a little hard to follow. Possible to make it a positive boolean check and then `if (!blah) return;`?

Reply:
Done.